### PR TITLE
Valid spawn locations

### PR DESCRIPTION
**Pull request by mes258**
Now the player will only spawn in a valid location after dying. Still need to make 2 changes before merging: 
- [x] Use Vincent's random chunk method from the random-respawn branch.

### CODE REVIEW

File: client/chunk.js [hunk 1]
```diff
@@ -3,11 +3,14 @@ var Chunk = function (position_, blocks_, colors_) {
   var position = position_.clone().floor();
   // assume blocks is a perfect X by Y by Z prism - validate this in future? TODO
   var blocks = blocks_;
+
   var colors = colors_;
   var graphicsObject = null;
 
   var blockUpdate = false;
 
+  var validSpawnLocations = null;
```
Comment (vinport16) [inline client/chunk.js]: initialize as empty list?

File: client/chunk.js [hunk 2]
```diff
@@ -206,6 +216,22 @@ var Chunk = function (position_, blocks_, colors_) {
     return Math.sqrt(sum);
   }
 
+  this.determineValidSpawnLocations = function(){
+    validSpawnLocations = []
+    for (let z = 0; z < blocks.length; z++) {
+      for (let y = 0; y < blocks[0].length; y++) {
+        for (let x = 0; x < blocks[0][0].length; x++) {
+          if(blocks[z][y][x] != null){
+            if(blocks[z][y][x] > 0 && (blocks[z][y+1] ==  null || blocks[z][y+1][x] == 0) && (blocks[z][y+2] == null || blocks[z][y+2][x] == 0)){
```
Comment (vinport16) [inline client/chunk.js:231]: when is a block null ?

File: client/world.js [hunk 3]
```diff
@@ -9,6 +9,8 @@ var World = function (chunkSize_, renderRadius_) {
   // how far do you have to move from the focused chunk to change focus?
   var chunkRadius = 2;
 
+  var worldSize = {x: 0, y: 0, z: 0}
```
Comment (vinport16) [inline client/world.js]: if u use the random chunk selection function I made, you don't need this anymore - I would say remove it

File: client/chunk.js (hunk 2 above)
Comment (mes258) [reply client/chunk.js:231]: Above the top of the map.

File: client/world.js (hunk 3 above)
Comment (mes258) [reply client/world.js]: Removed after adding your function.

**Pull request merged**